src/management-system-v2/components/properties-panel.tsx

```diff
@@ -2,206 +2,364 @@
 
 import type Modeling from 'bpmn-js/lib/features/modeling/Modeling';
 
+import { getFillColor, getStrokeColor } from 'bpmn-js/lib/draw/BpmnRenderUtil';
+
 import type { ElementLike } from 'diagram-js/lib/core/Types';
 
 import useModelerStateStore from '@/lib/use-modeler-state-store';
 
-import React, { FocusEvent, useEffect, useState } from 'react';
+import React, { FocusEvent, useCallback, useEffect, useMemo, useState } from 'react';
 
-import { Card, Input, ColorPicker, Drawer, Space, Image } from 'antd';
+import { Input, ColorPicker, Space, Image } from 'antd';
 
 import { EuroCircleOutlined, ClockCircleOutlined } from '@ant-design/icons';
+import BpmnFactory from 'bpmn-js/lib/features/modeling/BpmnFactory';
+import {
+  getMetaDataFromElement,
+  getMilestonesFromElement,
+  setProceedElement,
+} from '@proceed/bpmn-helper';
+import CustomPropertySection from './custom-property-section';
+import MilestoneSelectionSection from './milestone-selection-section';
+import ResizableElement from './ResizableElement';
+import CollapsibleCard from './collapsible-card';
 
 type PropertiesPanelProperties = {
   selectedElement: ElementLike;
   setOpen: (open: boolean) => void;
 };
 
 const PropertiesPanel: React.FC<PropertiesPanelProperties> = ({ selectedElement, setOpen }) => {
+  const [showInfo, setShowInfo] = useState(true);
+  const [width, setWidth] = useState(450);
+
+  const [metaData, setMetaData] = useState(getMetaDataFromElement(selectedElement.businessObject));
+  const [milestones, setMilestones] = useState(
+    getMilestonesFromElement(selectedElement.businessObject),
+  );
   const [name, setName] = useState('');
+  const [costsPlanned, setCostsPlanned] = useState('');
+  const [timePlannedDuration, setTimePlannedDuration] = useState('');
 
   const modeler = useModelerStateStore((state) => state.modeler);
 
+  const refreshMetaData = useCallback(() => {
+    setMetaData(getMetaDataFromElement(selectedElement.businessObject));
+  }, [selectedElement.businessObject]);
+
+  const refreshMilestones = useCallback(() => {
+    setMilestones(getMilestonesFromElement(selectedElement.businessObject));
+  }, [selectedElement.businessObject]);
+
+  useEffect(() => {
```

Comment:
unnecessary useEffect, metaData and milestones, could be instead defined with useMemo